File: Tests/c/six.c
#include <stdio.h>
#include <string.h>
#include <assert.h>

typedef struct {
    int id;
    char role[20];
} DatabaseRecord;

int check_admin_access(DatabaseRecord *record) {
    // Simulējam lomas iegūšanu no citas sistēmas daļas
    char required_role[20];
    strcpy(required_role, "ADMIN");

    // KĻŪDA: Tiek salīdzinātas rādītāju adreses, nevis virkņu saturs.
    // Šis nosacījums vienmēr būs nepatiess (false).
    if (record->role == required_role) {
        return 1;
    }
    return 0;
}

int main() {
    DatabaseRecord my_user = {1005, "ADMIN"};

    int has_access = check_admin_access(&my_user);

    assert(has_access == 1 && "Kļūda: Pieeja atteikta, kaut gan lomai jābūt ADMIN!");
    printf("Scenārijs 6 darbojas veiksmīgi.\n");
    return 0;
}

File: Tests/python/six.py
class DatabaseCache:

	def __init__(self):
		self._store = {
			1005: {"name": "Sistēmas administrators", "access": "ALL"},
			2048: {"name": "Viesis", "access": "READ_ONLY"}
		}
		self.queries_executed = 0

	def get_user_role(self, user_id: int):
		self.queries_executed += 1

		for cached_id, data in self._store.items():
			if user_id is cached_id:
				return data["access"]

		return "UNKNOWN"


if __name__ == "__main__":
	db = DatabaseCache()
	admin_id = int("100" + "5")

	role = db.get_user_role(admin_id)
	assert role == "ALL", f"Kļūda: Nevarēja atrast lomu ID {admin_id}. Iegūts: {role}"
	print("Scenārijs 6 darbojas veiksmīgi.")